File: android/src/main/kotlin/io/openim/flutter_openim_sdk_ffi/listener/OnListenerForService.kt
package io.openim.flutter_openim_sdk_ffi.listener

import io.openim.flutter_openim_sdk_ffi.models.FriendApplicationInfo
import io.openim.flutter_openim_sdk_ffi.models.GroupApplicationInfo
import io.openim.flutter_openim_sdk_ffi.models.Message

interface OnListenerForService {
    fun onFriendApplicationAccepted(u: FriendApplicationInfo?)
    fun onFriendApplicationAdded(u: FriendApplicationInfo?)
    fun onGroupApplicationAccepted(info: GroupApplicationInfo?)
    fun onGroupApplicationAdded(info: GroupApplicationInfo?)
    fun onRecvNewMessage(msg: Message?)
}

File: ios/Classes/listener/OnListenerForService.swift

protocol OnListenerForService {
    func onFriendApplicationAccepted(_ info: FriendApplicationInfo?)
    func onFriendApplicationAdded(_ info: FriendApplicationInfo?)
    func onGroupApplicationAccepted(_ info: GroupApplicationInfo?)
    func onGroupApplicationAdded(_ info: GroupApplicationInfo?)
    func onRecvNewMessage(_ msg: Message?)
}